### DISCUSSION TITLE
How to Control Passive Test Functions？

### DISCUSSION THREAD

**Discussion by Yingqingshan** [pytest-dev/pytest#11021] [General]
`import pytest

class TestUser:
    @pytest.mark.parametrize("username, password", [("admin", "123456"), ("admin1", "123456")])
    def test_login(self, username, password):
        print("login")

    @pytest.mark.parametrize("username, password", [("admin", "123456"), ("admin1", "123456")])
    def test_logout(self, username, password):
        print("logout")

    @pytest.mark.parametrize("username, password", [("admin", "123456"), ("admin1", "123456")])
    def test_register(self, username, password):
        print("register")

class TestExample:
    @pytest.mark.parametrize("username, password", [("admin", "123456"), ("admin1", "123456")])
    def test_order(self, username, password):
        TestUser().test_login(username, password)

if __name__ == '__main__':
    pytest.main()
`
How to control that when test_order calls the TestUser test class, only the called test_login test function is executed, and not all test functions under TestUser are executed? Is it possible to use the pytest_collection_modifyitems hook provided by pytest to control the test functions that do not need to execute all test classes?

Now the execution result is as follows：
![image](https://github.com/pytest-dev/pytest/assets/100674970/f3635162-ab70-471e-aac5-252e7b4a6846)

**Comment by RonnyPfannschmidt** [comment]
Im under the impression the ask here is for having dependent test cases, which intentionally is not supported as is

The recommendation is to have login state as a scoped fixture, so more than one integration test can use it safely
